src/software/ai/navigator/obstacle/obstacle_factory.h

```diff
@@ -94,29 +94,44 @@ class ObstacleFactory
      *
      * @return rectangular obstacle
      */
-    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle);
+    ObstaclePtr createObstacleFromRectangle(const Rectangle &rectangle) const;
 
    private:
     std::shared_ptr<const ObstacleFactoryConfig> config;
-    double shape_expansion_amount;
+    double obstacle_expansion_amount;
 
     /**
-     * Create circle obstacle directly from a Circle
-     * Note: this helper function does not add a robot radius
+     * Returns an obstacle for the shape expanded on all sides to account for the size of
+     * the robot
      *
-     * @param circle Circle
+     * @param The shape to expand
      *
-     * @return circle shaped obstacle
+     * @return expanded ObstaclePtr
      */
-    ObstaclePtr createObstacle(const Circle &circle);
+    ObstaclePtr expandForRobotSize(const Circle &circle) const;
+    ObstaclePtr expandForRobotSize(const Polygon &polygon) const;
 
     /**
-     * Create polygon obstacle directly from a Polygon
-     * Note: this helper function does not add a robot radius
+     * Returns an obstacle from the Rectangle expanded in the directions in vector
+     * multiplied to account for robot size
      *
-     * @param polygon Polygon
+     * @param rectangle Rectangle to expand
+     * @param directions The list of directions to expand the Rectangle
      *
-     * @return polygon shaped obstacle
+     * @return expanded Rectangle as an obstacle
      */
-    ObstaclePtr createObstacle(const Polygon &polygon);
+    ObstaclePtr expandForRobotSize(const Rectangle &rectangle,
```

Comment:
Upon looking at the usage of this function again I think the list of vectors is a bit confusing as an interface. Reading the consumer code, I immediately assumed things were expanded by the literal vectors (using their magnitudes) rather than the vectors just representing direction.

Because all we really care about is the direction, maybe a list of `Angle` would be better here.